### PR TITLE
Supermatter Can Display EER to Ghosts.

### PR DESCRIPTION
**Pull request by vincentiusvin**
## About The Pull Request
I'm not sure that there's a way for ghosts to check out the EER of the spicy rock without someone logging in on the console. This lets ghosts know if an SM is going to spawn tesla/skeleton in advance.

The part at the trait section (thanks to mothblocks) is changed to not cause runtimes when the supermatter is examined by ghosts.

## Why It's Good For The Game
Speculating on SM delamination event is fun.

## Changelog
:cl:
qol: You can now check the EER of the supermatter as a ghost by examining it.
/:cl:

### CODE REVIEW

**Review by LemonInTheDark — APPROVED** [review]
Good change, go with god

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -350,9 +350,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 /obj/machinery/power/supermatter_crystal/examine(mob/user)
 	. = ..()
-	var/immune = HAS_TRAIT(user, TRAIT_SUPERMATTER_MADNESS_IMMUNE) || HAS_TRAIT(user.mind, TRAIT_SUPERMATTER_MADNESS_IMMUNE)
+	var/immune = user?.mind && (HAS_TRAIT(user, TRAIT_SUPERMATTER_MADNESS_IMMUNE) || HAS_TRAIT(user.mind, TRAIT_SUPERMATTER_MADNESS_IMMUNE))
```
Comment (coiax) [inline code/modules/power/supermatter/supermatter.dm:353]:
```suggestion
	var/immune = HAS_TRAIT(user, TRAIT_SUPERMATTER_MADNESS_IMMUNE) || (user.mind && HAS_TRAIT(user.mind, TRAIT_SUPERMATTER_MADNESS_IMMUNE)))
```

The current line would mean that you wouldn't check if a mob was immune if they didn't have a mind?

@Mothblocks, this make sense?

**Comment by coiax** [comment]
Almost certainly out of scope, but if you could `attack_ghost` and get the supermatter TGUI program pop up, that would be even better.

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/power/supermatter/supermatter.dm:353]: they wouldnt be examining anyway, but sure

**Comment by vincentiusvin** [comment]
> Almost certainly out of scope, but if you could `attack_ghost` and get the supermatter TGUI program pop up, that would be even better.

That seems kinda cool, I will toy with the idea when I have time.

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -350,7 +350,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 /obj/machinery/power/supermatter_crystal/examine(mob/user)
 	. = ..()
-	var/immune = user?.mind && (HAS_TRAIT(user, TRAIT_SUPERMATTER_MADNESS_IMMUNE) || HAS_TRAIT(user.mind, TRAIT_SUPERMATTER_MADNESS_IMMUNE))
+	var/immune = HAS_TRAIT(user, TRAIT_SUPERMATTER_MADNESS_IMMUNE) || (user.mind && HAS_TRAIT(user.mind, TRAIT_SUPERMATTER_MADNESS_IMMUNE)))
```
Comment (LemonInTheDark) [inline code/modules/power/supermatter/supermatter.dm:353]: 3 ('s, 4 )'s

Comment (vincentiusvin) [reply code/modules/power/supermatter/supermatter.dm]: bruh

**Comment by LemonInTheDark** [comment]
Reminder to set the pr to Ready For Review when you're well, ready for review

**Comment by vincentiusvin** [comment]
This thing is ready alright, but I'm unable to make much progress on implementing the sm_monitor program datum into attack_ghost. Perhaps that should be attempted in another PR?

**Comment by LemonInTheDark** [comment]
It's up to you, this is pretty damn nice on its own. If you want to make this pr that, feel free, but you don't need to.

**Comment by vincentiusvin** [comment]
Mhm, i suppose the NT CIMS thing will have to wait then. This PR is ready.

**Pull request merged**